experimental/ActionPolyRing/src/Content.jl

```diff
@@ -847,6 +832,203 @@ function leader(apre::ActionPolyRingElem)
   return maximum(vars(apre))
 end
 
+###############################################################################
+#
+#  Discriminant and resultant
+#
+###############################################################################
+
+function resultant(r1::ActionPolyRingElem, r2::ActionPolyRingElem, var::ActionPolyRingElem)
+  check_parent(r1, r2) && check_parent(r1, var)
+  @req is_gen(var) "Not a jet variable"
+  return resultant(r1, r2, __vtj(parent(var))[var])
+end
+
+@doc raw"""
+    resultant(f::ActionPolyRingElem, g::ActionPolyRingElem, i::Int, jet::Vector{Int})
+
+Return the resultant of `f` and `g` regarded as univariate polynomials in the jet variable specified by `i` and
+`jet`. This method allows all versions described in [Specifying jet variables](@ref specifying_jet_variables).
+"""
+function resultant(r1::ActionPolyRingElem, r2::ActionPolyRingElem, i::Int, jet::Vector{Int})
+  check_parent(r1, r2)
+  S = parent(r1)
+  @req __is_valid_jet(S, i, jet) "Invalid jet variable"
+  jtv = __jtv(S)
+  if haskey(jtv, (i, jet))
+    return S(resultant(data(r1), data(r2), data(jtv[(i, jet)])))
+  end
+  if is_zero(r1) || is_zero(r2)
+    return zero(S)
+  end
+  return one(S)
+end
+
+resultant(r1::ActionPolyRingElem, r2::ActionPolyRingElem, jet_idx::Tuple{Int,Vector{Int}}) = resultant(r1, r2, jet_idx...)
+resultant(r1::ActionPolyRingElem, r2::ActionPolyRingElem, i::Int) = resultant(r1, r2, gen(parent(r1), i))
+
+@doc raw"""
+    discriminant(p::ActionPolyRingElem)
+
+Return the discriminant of `p`.
+"""
+function discriminant(p::ActionPolyRingElem)
+  if is_constant(p)
+    return zero(parent(p))
+  end
+
+  ld = leader(p)
+  
+  if degree(p, ld) % 4 in [0,1]
```

Comment:
```suggestion
  if degree(p, ld) % 4 in (0,1)
```
for slightly better code